ftrip.io.catalog-service/ftrip.io.catalog-service/Accommodations/AccommodationRepository.cs

```diff
@@ -24,7 +32,123 @@ public override async Task<Accommodation> Read(Guid id, CancellationToken cancel
                 .Include(a => a.PropertyType)
                 .Include(a => a.Amenities).ThenInclude(aa => aa.Amenity)
                 .Include(a => a.Location)
+                .Include(a => a.Availabilities)
+                .Include(a => a.PriceDiffs)
                 .FirstOrDefaultAsync(e => e.Id.Equals(id), cancellationToken);
         }
+
+        public Task<Accommodation> ReadSimple(Guid id, CancellationToken ct = default) => base.Read(id, ct);
+
+        public async Task<Accommodation> Update(UpdateAccommodationRequest accommodationUpdate, CancellationToken ct = default)
+        {
+            var accommodation = await Read(accommodationUpdate.Id, ct);
+            if (accommodation != null)
+            {
+                _context.Entry(accommodation).CurrentValues.SetValues(accommodationUpdate);
+            }
+            return accommodation;
+        }
+        public async Task<Accommodation> Update(UpdateAccommodationLocationRequest accommodationUpdate, CancellationToken ct = default)
+        {
+            var accommodation = await Read(accommodationUpdate.Id, ct);
+            if (accommodation != null)
+            {
+                _context.Entry(accommodation.Location).CurrentValues.SetValues(accommodationUpdate.Location);
+            }
+            return accommodation;
+        }
+        public async Task<Accommodation> Update(UpdateAccommodationAmenitiesRequest accommodationUpdate, CancellationToken ct = default)
+        {
+            var accommodation = await Read(accommodationUpdate.Id, ct);
+            if (accommodation == null) return null;
+            foreach (var existingAccommodationAmenity in accommodation.Amenities)
```

Comment:
You have very similar login in few of the methods. Extract removed, changed and existing. What I would do here is create new class like

class ListsDiff<T> w
here T : IEntity {
List<T> Removed, List<T> Added, List<T> Updated
}

And then implement some method like:
ListsDiff<T> Diff<T>(List<T> old, List<T> new)
which will do this logic.

That will allow you to use Add/RemoveRange method and will be a lot easier to do this logic in the future.